R/class-color.R

```diff
@@ -35,4 +34,51 @@ wb_colour <- function(
   z
 }
 
+#' @export
+#' @rdname wbColour
+#' @usage NULL
+wb_color <- wb_colour
+
 is_wbColour <- function(x) inherits(x, "wbColour")
+
+#' takes color and returns colour
+#' @param ... ...
+#' @returns named colour argument
+#' @keywords internal
+#' @noRd
+standardise_color_names <- function(...) {
```

Comment:
Do I understand this correctly?

```r
foo <- function(color, ...) {
  standardized_color_name(...)  
  color
}

foo(color = 1)
#> [1] 1

foo(colour = 1)
#> [1] 1
```

Not sure how much I like messing around with re-assigning values.  A simpler implementation could be:

``` r
foo <- function(color = colour, colour) {
  check_color_colour(color, colour)
  color
}

check_color_colour <- function(color, colour) {
  if (missing(colour)) {
    return(invisible())
  }
  
  if (!identical(color, colour)) {
    stop("both `color` and `colour` cannot be set")
  }
}

# setting only color
foo(1)
#> [1] 1

# setting only colour (turns into color)
foo(colour = 1)
#> [1] 1

# both are identical, that's fine
foo(color = 1, colour = 1)
#> [1] 1

# difference, we have a problem
try(foo(color = 1, colour = 2)) 
#> Error in check_color_colour(color, colour) : 
#>   both `color` and `colour` cannot be set
```

<sup>Created on 2022-12-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
We'd have to include `@params color, colour` but it would keep it explicitly clear to the user.

Reply:
1) Yes, that is all it does.
2) I don't like duplicating all arguments. It creates visual confusion, especially with the border stuff and all the different colors. We'll end up with 10 different color arguments and a bunch of comparison functions. Therefore I mimicked the `ggplot2::aes()` approach. The difference to their approach is that they only have a single argument to care about, while we have various. But true, I haven't considered the `foo(color = 1, colour = 2)` approach. Might want to throw a warning if this exists in `parent.frame()`

Reply:
Update: 12 different colors ... we have two for the inner grid and checking with `exists()` is not enough,  because they all exist at this stage. Maybe this will change once you get to #226 *\*winkwink\**.
Therefore I consider keeping it the way it is. After all no harm is done this way. If the user ends up with a unexpected color ... there are more important things to worry about. If their input creates no color at all, fine too. I'd love for R to have a dynamic way to handle this case. Otherwise I simply strive for a consistent user interface.